app/assets/javascripts/discourse/controllers/add-post-notice.js.es6

```diff
@@ -0,0 +1,38 @@
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import computed from "ember-addons/ember-computed-decorators";
+
+export default Ember.Controller.extend(ModalFunctionality, {
+  notice: null,
+  saving: false,
+
+  @computed("saving", "notice")
+  disabled(saving, notice) {
+    return saving || Ember.isEmpty(notice);
+  },
+
+  onShow() {
+    this.setProperties({
+      notice: "",
+      saving: false
+    });
+  },
+
+  actions: {
+    setNotice() {
+      this.set("saving", true);
+
+      const post = this.get("post");
+      const notice = this.get("notice");
+      const resolve = this.get("resolve");
+
+      post.updatePostField("notice", notice).then(() => {
+        post.setProperties({
+          notice_type: "custom",
+          notice_args: notice
+        });
+        resolve();
```

Comment:
What if the `updatePostField` promise fails? `resolve` will never be called. It should probably be in a `finally` method.

Either that or we should be passing in the `reject` method and calling that from `catch`.

Additionally, if the modal is closed, we'll want to call `resolve` too so we aren't leaking a promise that way.